symengine/pow.cpp

```diff
@@ -164,6 +183,39 @@ RCP<const Basic> pow(const RCP<const Basic> &a, const RCP<const Basic> &b)
         RCP<const Pow> A = rcp_static_cast<const Pow>(a);
         return pow(A->get_base(), mul(A->get_exp(), b));
     }
+    if (eq(*a, *E)) {
+        if (is_a<Mul>(*b) and exp_mul_helper(b)) {
+            RCP<const Number> coef = down_cast<const ComplexBase &>(
+                                         *down_cast<const Mul &>(*b).get_coef())
+                                         .imaginary_part();
+            RCP<const Basic> arg = mul(coef, integer(2));
+            long n = ((down_cast<const Integer &>(*arg).as_int() % 4) + 4) % 4;
+            if (!n) {
+                return one;
+            } else if (n == 1) {
+                return I;
+            } else if (n == 2) {
+                return minus_one;
+            } else {
+                return Complex::from_two_nums(*zero, *minus_one);
+            }
+        } else if (is_a<Add>(*b)) {
+            const umap_basic_num &dict = down_cast<const Add &>(*b).get_dict();
+            umap_basic_num new_dict;
+            RCP<const Number> coef = down_cast<const Add &>(*b).get_coef();
+            RCP<const Basic> s = one;
+            for (const auto &p : dict) {
+                if (eq(*p.first, *pi)
+                    and exp_mul_helper(mul(p.first, p.second))) {
+                    s = exp(mul(p.first, p.second));
+                } else {
+                    Add::dict_add_term(new_dict, p.second, p.first);
+                }
+            }
+            return mul(s, make_rcp<const Pow>(
+                              a, Add::from_dict(coef, std::move(new_dict))));
+        }
+    }
```

Comment:
We need to carefully benchmark these two if branches, before and after this PR. So the first if clause would run for something like `E^(a*b)`. The second one for `E^(a+b)`. Correct?

If so, then we need to benchmark these two things and similar things a lot.